scripts/quests/sandoria/Grave_Concerns.lua

```diff
@@ -0,0 +1,208 @@
+-----------------------------------
+-- Grave_Concerns
+-----------------------------------
+-- Log ID: 0, Quest ID: 11
+-- !addquest 0 11
+-- !additem 547 -- tomb guards waterskin
+-- !additem 567 -- skin of well water
+-- Andecia : !pos 167 0 45 230
+-- Tombstone_Upper : !pos 1 0.1 -101 190
+-- Well : !pos -129 -6 92 230
+-----------------------------------
+local sandoriaID = zones[xi.zone.SOUTHERN_SAN_DORIA]
+local krtID = zones[xi.zone.KING_RANPERRES_TOMB]
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.SANDORIA, xi.quest.id.sandoria.GRAVE_CONCERNS)
+
+quest.reward =
+{
+    fameArea = xi.fameArea.SANDORIA,
+    gil = 560,
+    title = xi.title.ROYAL_GRAVE_KEEPER,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getFameLevel(xi.fameArea.SANDORIA) >= 1
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Andecia'] = function(player, npc)
+                if quest:getVar(player, 'Introduction') == 0 then
+                    return quest:progressEvent(540)
+                elseif quest:getVar(player, 'Introduction') == 1 then
+                    return quest:progressEvent(541)
+                end
+            end,
+
+            onEventFinish =
+            {
+                [540] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Introduction', 1)
+                end,
+
+                [541] = function(player, csid, option, npc)
+                    if option == 0 then
+                        if player:getFreeSlotsCount() == 0 then
+                            player:messageSpecial(sandoriaID.text.ITEM_CANNOT_BE_OBTAINED, 567) -- Well Water
+                        else
+                            quest:begin(player)
+                            player:addItem(xi.item.SKIN_OF_WELL_WATER)
+                            player:messageSpecial(sandoriaID.text.ITEM_OBTAINED, xi.item.SKIN_OF_WELL_WATER) -- Well Water
```

Comment:
Take a look at https://github.com/LandSandBoat/server/blob/base/scripts/globals/npc_util.lua#L224

We would want to use npcUtil as it will handle all that fun stuff.